lib/trino-parquet/src/main/java/io/trino/parquet/ParquetReaderUtils.java

```diff
@@ -29,4 +32,81 @@ public static ByteBufferInputStream toInputStream(DictionaryPage page)
     {
         return toInputStream(page.getSlice());
     }
+
+    /**
+     * Reads an integer formatted in ULEB128 variable-width format described in
+     * <a href="https://en.wikipedia.org/wiki/LEB128">...</a>
+     */
+    public static int readUleb128Int(SimpleSliceInputStream input)
+    {
+        byte[] inputBytes = input.getByteArray();
+        int offset = input.getByteArrayOffset();
+        // Manual loop unrolling shows improvements in BenchmarkReadUleb128Int
```

Comment:
What's the performance difference? The downside is that it's more bytecode and fewer opportunities for inlining as a result.

Reply:
Below are JMH results comparing this with a simple while loop implementation
```
Benchmark                                   (maxValue)  (size)   Mode  Cnt     Score    Error   Units
BenchmarkReadUleb128Int.readUleb128Int           20000     100  thrpt   45  2630.568 ± 60.622  ops/ms
BenchmarkReadUleb128Int.readUleb128Int           20000    1000  thrpt   45   182.463 ±  5.855  ops/ms
BenchmarkReadUleb128Int.readUleb128Int         4000000     100  thrpt   45  1958.759 ± 55.983  ops/ms
BenchmarkReadUleb128Int.readUleb128Int         4000000    1000  thrpt   45   171.086 ±  6.803  ops/ms
BenchmarkReadUleb128Int.readUleb128IntLoop       20000     100  thrpt   45  1862.733 ± 47.458  ops/ms
BenchmarkReadUleb128Int.readUleb128IntLoop       20000    1000  thrpt   45   146.719 ±  5.720  ops/ms
BenchmarkReadUleb128Int.readUleb128IntLoop     4000000     100  thrpt   45  1208.188 ± 40.338  ops/ms
BenchmarkReadUleb128Int.readUleb128IntLoop     4000000    1000  thrpt   45    96.781 ±  3.187  ops/ms
```
I'm updating BenchmarkReadUleb128Int to include runs with the while loop implementation for future reference.